core/src/main/java/com/datastrato/gravitino/authorization/PermissionManager.java

```diff
@@ -67,9 +67,12 @@ User grantRolesToUser(String metalake, List<String> roles, String user) {
           UserEntity.class,
           Entity.EntityType.USER,
           userEntity -> {
-            List<RoleEntity> roleEntities =
-                roleManager.getValidRoles(metalake, userEntity.roleNames(), userEntity.roleIds());
-
+            List<RoleEntity> roleEntities = Lists.newArrayList();
+            if (userEntity.roleNames() != null) {
+              for (String role : userEntity.roleNames()) {
+                roleEntities.add(roleManager.getRole(metalake, role));
+              }
+            }
```

Comment:
I would suggest that we don't store the role name in user entity, only store the id, and we don't need to check role name, but the role id.

Reply:
We don't store role name in the underlying relational database. It's necessary for us to  assign the role names fields in the user entity.  The updated user entity will be returned to end user. The users of requests need role names for human understanding.